### DISCUSSION TITLE
Building an easy consult-bbdb

### DISCUSSION THREAD

**Discussion by WorldsEndless** [minad/consult#926] [Q&A] [answered]
BBDB has the command `(bbdb)`, which does a good thing: it takes an input, takes an "RET" key, then searches the bbdb database, and returns a list of results. It would be nice to have Consult in the equation to get those search results as-I-type. The end goal for me is to view the results so I can see details on some BBDB contact without that extra RET key-press. Is there a simple way to wire up a consult wrapper for the bbdb command or some other strategy to simply get at those results?

**Comment by minad** [comment, marked as answer]
I am not familiar with bbdb so I cannot help you with the details. I suggest you take a look at `consult--dynamic-collection` which allows you to dynamically generate candidates. `consult--dynamic-collection` is used by `consult-line-multi` and `consult-info` to find matching lines in buffers/info files. In a similar way, you could scan the bbdb database. I hope this helps!

**Comment by Scaalp-fr** [comment]
Hello,

I wanted to complete my email addresses from the bbdb, so I wrote `cape-bbdb` to integrate the bbdb with corfu.  This is not for consult, but perhaps it could still help you.  It shows the complete bbdb entry when you select one in the list.  The code in on codeberg.
